python/cudf/cudf/tests/test_udf_masked_ops.py

```diff
@@ -586,3 +582,39 @@ def func(row):
 
     data = cudf.DataFrame(data)
     run_masked_udf_test(func, func, data)
+
+
+@pytest.mark.parametrize(
+    "unsupported_dtype",
+    ["string", "decimal", "categorical", "interval", "list", "struct"],
+)
+def test_masked_udf_unsupported_dtype(unsupported_dtype):
+    def func(row):
+        return row[unsupported_dtype]
+
+    data = cudf.DataFrame()
+    data["string"] = ["a", "b", "c"]
+    data["decimal"] = _decimal_series(
+        ["1.0", "2.0", "3.0"], dtype=cudf.Decimal64Dtype(2, 1)
+    )
+    data["categorical"] = cudf.Series([1, 2, 3], dtype="category")
+    data["interval"] = cudf.interval_range(start=0, end=3, closed=True)
+    data["list"] = [[1, 2], [3, 4], [5, 6]]
+    data["struct"] = [{"a": 1}, {"a": 2}, {"a": 3}]
```

Comment:
It feels a little odd to parametrize on the unsupported dtype but construct a DataFrame with all the columns every time. Could we parametrize on the data as well and just have a single column DataFrame tested each time?

Reply:
Ah you're right. Changed this so that the unsupported column is itself the parameter.